File: freecodecamp/daily-code-challenge/2026-03-06.js
function navigateTrail(map) {
  const rows = map.length;
  const cols = map[0].length;

  const grid = map.map(r => r.split(""));

  let start;

  for (let r = 0; r < rows; r++) {
    for (let c = 0; c < cols; c++) {
      if (grid[r][c] === "C") start = [r, c];
    }
  }

  const dirs = [
    [0, 1, "R"],
    [1, 0, "D"],
    [0, -1, "L"],
    [-1, 0, "U"]
  ];

  let [r, c] = start;
  let prev = null;
  let result = "";

  while (grid[r][c] !== "G") {
    for (let [dr, dc, move] of dirs) {
      let nr = r + dr;
      let nc = c + dc;

      if (
        nr >= 0 && nr < rows &&
        nc >= 0 && nc < cols &&
        (grid[nr][nc] === "T" || grid[nr][nc] === "G") &&
        (!prev || nr !== prev[0] || nc !== prev[1])
      ) {
        prev = [r, c];
        r = nr;
        c = nc;
        result += move;
        break;
      }
    }
  }

  return result;
}

File: freecodecamp/daily-code-challenge/2026-03-06.py
def navigate_trail(trail_map):
    rows = len(trail_map)
    cols = len(trail_map[0])

    grid = [list(row) for row in trail_map]

    start = None
    for r in range(rows):
        for c in range(cols):
            if grid[r][c] == "C":
                start = (r, c)

    directions = [
        (0, 1, "R"),
        (1, 0, "D"),
        (0, -1, "L"),
        (-1, 0, "U")
    ]

    r, c = start
    prev = None
    result = ""

    while grid[r][c] != "G":
        for dr, dc, move in directions:
            nr, nc = r + dr, c + dc

            if (0 <= nr < rows and 0 <= nc < cols and
                grid[nr][nc] in ("T", "G") and
                prev != (nr, nc)):

                prev = (r, c)
                r, c = nr, nc
                result += move
                break

    return result
